src/components/context/QuizContext.tsx

```diff
@@ -57,16 +57,16 @@ export const QuizProvider = ({ children }: Props) => {
     setCurrentIdx(computedIdx);
   };
 
-  const deleteQuestion = (questionId: string) => () => {
+  const deleteQuestion = (question: QuestionDataAppSettingRecord) => () => {
     // update list order
     const newOrder = order.toJS();
-    const idx = order.findIndex((id) => id === questionId);
+    const idx = order.findIndex((id) => id === question?.data?.questionId);
     newOrder.splice(idx, 1);
     if (orderSetting) {
       patchAppSetting({ id: orderSetting.id, data: { list: newOrder } });
     }
     // delete question
-    deleteAppSetting({ id: questionId });
+    deleteAppSetting({ id: question?.id });
```

Comment:
In your function definition `question` is not optional so is supposed to exist (aka you would not need `?.`). But I guess that on runtime you had errors so you added `?.` for it to work. In that case you should update to
```js
const deleteQuestion = (question?: QuestionDataAppSettingRecord) => () => {
```
and in that case maybe really handle empty cases (eg. you should not try to delete something undefined)